petab/core.py

```diff
@@ -900,3 +900,99 @@ def measurements_have_replicates(measurement_df: pd.DataFrame) -> bool:
             measurement_df,
             ['observableId', 'simulationConditionId',
              'preequilibrationConditionId', 'time'])).size().values - 1)
+
+
+def flatten_timepoint_specific_output_overrides(
+        petab_problem: Problem) -> None:
+    """If the PEtab problem definition has timepoint-specific
+    observableParameters or noiseParameters for the same observable, replace
+    those by replicating the respective observable.
+
+    This is a helper function for some tools which may not support such
+    timepoint-specific mappings.
+
+    Arguments:
+        petab_problem:
+            PEtab problem to work on
+    """
+
+    # Create empty df -> to be filled with replicate-specific observables
+    df_new = pd.DataFrame()
+
+    # Get observableId, preequilibrationConditionId
+    # and simulationConditionId columns in measurement df
+    df = petab_problem.measurement_df[
+        ["observableId",
+         "preequilibrationConditionId",
+         "simulationConditionId"]
+    ]
+    # Get unique combinations of observableId, preequilibrationConditionId
+    # and simulationConditionId
+    df_unique_values = df.drop_duplicates()
+
+    # Loop over each unique combination
+    for nrow in range(len(df_unique_values.index)):
+        df = petab_problem.measurement_df.loc[
+            (petab_problem.measurement_df['observableId'] ==
+             df_unique_values.loc[nrow, "observableId"])
+            & (petab_problem.measurement_df['preequilibrationConditionId'] <=
+               df_unique_values.loc[nrow, "preequilibrationConditionId"])
+            & (petab_problem.measurement_df['simulationConditionId'] <=
+               df_unique_values.loc[nrow, "simulationConditionId"])
+        ]
+
+        # Get list of unique observable parameters
+        unique_sc = df["observableParameters"].unique()
+        # Get list of unique noise parameters
+        unique_noise = df["noiseParameters"].unique()
+
+        # Loop
+        for n_noise in range(len(unique_noise)):
+            for n_scale in range(len(unique_sc)):
+                # Find the position of all instances of n_noise and n_scale
+                # in their corresponding column
+                # (full-string matches are denoted by zero)
+                idxs = (
+                        df["noiseParameters"].str.find(unique_noise[n_noise]) +
+                        df["observableParameters"].str.find(unique_sc[n_scale])
+                )
+                tmp = df.loc[idxs == 0, "observableId"]
+                # Update the observable name to replicate-specific
+                df.loc[idxs == 0, "observableId"] = tmp+"_"+str(
+                    n_scale + n_noise + 1
```

Comment:
will this really be replicate-specific? a + b + 1 --> couldn't it be that you have obs parameters and 2 noise parameters, such that e.g. 1 + 0 + 1 = 0 + 1 + 1? maybe rather ``tmp + '_' + str(n_scale) + '_' + str(n_noise)``?

Reply:
actually a new replicate should have to appear, e.g., `obs1_4`. I will address this.

Reply:
this is already covered in the test case:
`1         obs1            condition1  ...      obsParOverride2  noiseParOverride1`
`2         obs1            condition1  ...      obsParOverride2  noiseParOverride2`

same scaling, but different sigma